eslint.config.mjs

```diff
@@ -0,0 +1,155 @@
+import { defineConfig, globalIgnores } from 'eslint/config';
+import { FlatCompat } from '@eslint/eslintrc';
+import js from '@eslint/js';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+import reactHooks from 'eslint-plugin-react-hooks';
+import importPlugin from 'eslint-plugin-import';
+
+import tsParser from '@typescript-eslint/parser';
+import typescriptEslint from '@typescript-eslint/eslint-plugin';
+import stylistic from '@stylistic/eslint-plugin';
+
+import prettier from 'eslint-plugin-prettier';
+import jsdoc from 'eslint-plugin-jsdoc';
+import globals from 'globals';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const compat = new FlatCompat({
+  baseDirectory: __dirname,
+  recommendedConfig: js.configs.recommended,
+  allConfig: js.configs.all,
```

Comment:
I'd like to confirm something. In original eslintrc we have the list of recommended configs being extended:
```
"extends": [
    "eslint:recommended",
    "plugin:jsdoc/recommended",
    "prettier",
    "prettier/@typescript-eslint"
  ],
```
How are these being replaced in flat config?